src/apprt/gtk/GlobalShortcuts.zig

```diff
@@ -0,0 +1,412 @@
+const GlobalShortcuts = @This();
+
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+
+const gio = @import("gio");
+const glib = @import("glib");
+const gobject = @import("gobject");
+
+const App = @import("App.zig");
+const configpkg = @import("../../config.zig");
+const Binding = @import("../../input.zig").Binding;
+const key = @import("key.zig");
+
+const log = std.log.scoped(.global_shortcuts);
+const Token = [16]u8;
+
+app: *App,
+arena: std.heap.ArenaAllocator,
+dbus: *gio.DBusConnection,
+
+/// A mapping from a unique ID to an action.
+/// Currently the unique ID is simply the serialized representation of the
+/// trigger that was used for the action as triggers are unique in the keymap,
+/// but this may change in the future.
+map: std.StringHashMapUnmanaged(Binding.Action) = .{},
```

Comment:
Should this be an ArrayHashMap to avoid the tombstone issue for very long running Ghostty instances? https://github.com/ziglang/zig/issues/17851

Reply:
Yeah. In this case it'd be pretty pathological because you only clear and re-add when settings are reloaded so a user would have to reload _a lot_ but I've been bit enough by this bug that I instinctively think "why not"